src/Pact/Analyze/Translate.hs

```diff
@@ -447,80 +470,102 @@ translateBody = \case
       ESimple ty astsT -> ESimple ty $ Sequence ast' astsT
       EObject ty astsO -> EObject ty $ Sequence ast' astsO
 
+translateLet :: ScopeType -> [(Named Node, AST Node)] -> [AST Node] -> TranslateM ETerm
+translateLet scopeTy (unzip -> (bindingAs, rhsAs)) body = do
+  bindingTs <- traverse translateBinding bindingAs
+  rhsETs <- traverse translateNode rhsAs
+
+  retTid <- genTagId
+
+  let wrapWithLets :: Term a -> Term a
```

Comment:
code this twisty needs a comment. also, you sure this couldn't be a `foldr`?

Reply:
I'll add a comment to clarify intent.

Re: `foldr`, we could use either here, but the rule of thumb I follow is that unless the ordering dictates `foldr` or one would like the option to exit the fold early, one should just default to strict `foldl'`.